### ISSUE TITLE
More useful error message if enable_load_extension is not available

### ISSUE THREAD

**Issue by simonw** [simonw/datasette#1979]
I get this from:

    datasette --load-extension spatialite --get /-/versions.json

```
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/datasette/app.py", line 614, in _prepare_connection
    conn.enable_load_extension(True)
AttributeError: 'sqlite3.Connection' object has no attribute 'enable_load_extension'
```
It would be useful if Datasette caught this error and output something more friendly.

**Comment by simonw** [comment]
This helps:
```diff
diff --git a/datasette/cli.py b/datasette/cli.py
index 2b61292b..ea98879c 100644
--- a/datasette/cli.py
+++ b/datasette/cli.py
@@ -4,13 +4,15 @@ import click
 from click import formatting
 from click.types import CompositeParamType
 from click_default_group import DefaultGroup
+import functools
 import json
 import os
 import pathlib
 import shutil
 from subprocess import call
-import sys
 from runpy import run_module
+import sys
+import textwrap
 import webbrowser
 from .app import (
     OBSOLETE_SETTINGS,
@@ -126,7 +128,7 @@ class Setting(CompositeParamType):
 
 
 def sqlite_extensions(fn):
-    return click.option(
+    fn = click.option(
         "sqlite_extensions",
         "--load-extension",
         type=LoadExtension(),
@@ -134,6 +136,25 @@ def sqlite_extensions(fn):
         multiple=True,
         help="Path to a SQLite extension to load, and optional entrypoint",
     )(fn)
+    # Wrap it in a custom error handler
+    @functools.wraps(fn)
+    def wrapped(*args, **kwargs):
+        try:
+            return fn(*args, **kwargs)
+        except AttributeError as e:
+            if "enable_load_extension" in str(e):
+                raise click.ClickException(
+                    textwrap.dedent(
+                        """
+                    Your Python installation does not have the ability to load SQLite extensions.
+
+                    More information: https://docs.datasette.io/en/stable/installation.html#extensions
+                    """
+                    ).strip()
+                )
+            raise
+
+    return wrapped
 
 
 @click.group(cls=DefaultGroup, default="serve", default_if_no_args=True)
```
Need to write help for that to link to.

**Comment by simonw** [comment]
Once I ship the next release I should change that `/help/extensions` link to go to `/stable/` and not `/latest/`.

**Comment by simonw** [comment]
https://datasette.io/help/extensions now redirects to this new section of documentation: https://docs.datasette.io/en/latest/installation.html#a-note-about-extensions

**Comment by simonw** [comment]
```
% datasette --load-extension foo                        
Error: Your Python installation does not have the ability to load SQLite extensions.

More information: https://datasette.io/help/extensions
```

**Comment by simonw** [comment]
I back-ported this fix to `0.63.x` as well.
